photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/training/Driver.scala

```diff
@@ -514,18 +551,22 @@ final class Driver(val params: Params, val sparkContext: SparkContext, val logge
 
     trainingDataSet.foreach { case (_, rddLike: RDDLike) => rddLike.unpersistRDD() }
 
+    timer.start()
+    val bestModel = selectBestModel(gameModelsMap, validatingDataAndEvaluatorsOption)
+    timer.stop()
+    logger.info(s"Time elapsed after selecting best model: ${timer.durationSeconds} (s)\n")
+
```

Comment:
IMHO, selecting the best model is akin to tuning hyper-parameters on a validation data set, which logically does not belong with "saving to HDFS". I would like to keep those 2 features/concerns separate. Furthermore, if running the algorithm without "saving to HDFS" at the end is useful to e.g. track what the algo is doing via logs, or to look at an evaluation of the model, by that same token, users could be interested in seeing what happens during model selection. It could also be a diagnosis tool for us.